### PR TITLE
Fix 'Recommended extensions' test: timeouts and installed extension menu item

### PR DESCRIPTION
**Pull request by dmytro-ndp**
### What does this PR do?
1) Fixes timeout parameters which `Workspaces.waitWorkspaceListItemAbcence()` method is using to call `DriverHelper.waitDisappearance()` method
2) Refresh editor after extension is installed, to complete installation of extensions which require editor reload:
Screenshot from 2023-04-22 15-16-22
screenshot-Check_if_extensions_is_installed_and_enabled

### What issues does this PR fix or reference?
"**RecommendedExtentions**" test waited 20 minutes on cpp workspace item disappearance in Dashboard when remove test workspace: https://main-jenkins-csb-crwqe.apps.ocp-c1.prod.psi.redhat.com/job/Testing/job/e2e/job/basic/job/typescript-tests/7506/

"**RecommendedExtentions**" test failed on waiting "Disable" menu item is available: https://main-jenkins-csb-crwqe.apps.ocp-c1.prod.psi.redhat.com/job/Testing/job/e2e/job/basic/job/typescript-tests/7524/console

Related issue: https://issues.redhat.com/browse/CRW-4129

-------------------

PR check:
- [C/C++ recommended extensions run](https://main-jenkins-csb-crwqe.apps.ocp-c1.prod.psi.redhat.com/job/Testing/job/e2e/job/basic/job/typescript-tests/7581/console)

- ['Java 11 with Lombok' RecomendedExtentions run looking for javascript](https://main-jenkins-csb-crwqe.apps.ocp-c1.prod.psi.redhat.com/job/Testing/job/e2e/job/basic/job/typescript-tests/7650/console).
- ['Regular Java 11 with Lombok' RecomendedExtentions run](https://main-jenkins-csb-crwqe.apps.ocp-c1.prod.psi.redhat.com/job/Testing/job/e2e/job/basic/job/typescript-tests/7651/).

### PR Checklist
[As the author of this Pull Request I made sure that:](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#pull-request-template-and-its-checklist)

- [x] [The Eclipse Contributor Agreement is valid](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-eclipse-contributor-agreement-is-valid)
- [x] [Code produced is complete](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-produced-is-complete)
- [x] [Code builds without errors](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-builds-without-errors)
- [ ] [Tests are covering the bugfix or new feature ](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#tests-are-covering-the-bugfix-or-new-feature)
- [ ] [The repository devfile is up to date and works](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-repository-devfile-is-up-to-date-and-works)
- [ ] [Sections `What issues does this PR fix or reference` and `How to test this PR` completed](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#sections-what-issues-does-this-pr-fix-or-reference-and-how-to-test-this-pr-completed)
- [ ] [Relevant user documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [ ] [Relevant contributing documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [ ] [CI/CD changes implemented, documented and communicated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#cicd-changes-implemented-documented-and-communicated)

### Reviewers

Reviewers, please comment how you tested the PR when approving it.

### CODE REVIEW

**Comment by dmytro-ndp** [comment]
@nallikaea : indeed, extension with both hidden "Enable" and "Disable" items requires reloading of workspace page to finish installation:
 Screenshot from 2023-04-24 10-29-14

And I have just realized, that test is looking for a wrong extension when filter on "java" for "redhat.java" extensions name
_Actual_: "JavaScript Debugger"
_Expected_: "Language Support for Java(TM) by Red Hat"
Screenshot from 2023-04-24 10-32-38

**Comment by dmytro-ndp** [comment]
@nallikaea : please, review fixed PR where refrseh page step is added to complete extension installation:
['Java 11 with Lombok' RecomendedExtentions run looking for javascript](https://main-jenkins-csb-crwqe.apps.ocp-c1.prod.psi.redhat.com/job/Testing/job/e2e/job/basic/job/typescript-tests/7650/console).

['Regular Java 11 with Lombok' RecomendedExtentions run](https://main-jenkins-csb-crwqe.apps.ocp-c1.prod.psi.redhat.com/job/Testing/job/e2e/job/basic/job/typescript-tests/7651/).

File: tests/e2e/specs/dashboard-samples/RecomendedExtentions.spec.ts [hunk 1]
```diff
@@ -102,38 +104,51 @@ suite(`Check if recommended extensions installed for ${samples}`, async function
...
             extensionsView = await (await new ActivityBar().getViewControl('Extensions'))?.openView();
-            Logger.debug(`extensionsView?.getContent().getSections(): get current section.`);
-            [extensionSection] = await extensionsView?.getContent().getSections() as ExtensionsViewSection[];
         });
 
         test(`Wait until extensions starts installation`, async function (): Promise<void> {
             Logger.info(`Time for extensions installation TimeoutConstants.TS_COMMON_PLUGIN_TEST_TIMEOUT=${TimeoutConstants.TS_COMMON_PLUGIN_TEST_TIMEOUT}`);
             await driverHelper.wait(TimeoutConstants.TS_COMMON_PLUGIN_TEST_TIMEOUT);
+            browserTabsUtil.refreshPage();
+            await driverHelper.wait(TimeoutConstants.TS_OPEN_EDITOR_TIMEOUT);
```
Comment (nallikaea) [inline tests/e2e/specs/dashboard-samples/RecomendedExtentions.spec.ts]: can we use  explicit wait, something like `await projectAndFileTests.waitWorkspaceReadinessForCheCodeEditor()` ?

Comment (dmytro-ndp) [reply tests/e2e/specs/dashboard-samples/RecomendedExtentions.spec.ts]: Thanks for the hint!
Fixed.

New successful run: https://main-jenkins-csb-crwqe.apps.ocp-c1.prod.psi.redhat.com/job/Testing/job/e2e/job/basic/job/typescript-tests/7652/console

**Review by nallikaea — APPROVED** [review]

**Review by ScrewTSW — APPROVED** [review]
LGTM

**Pull request merged**